core/networkserver/server.go

```diff
@@ -97,8 +97,8 @@ func (s *networkServerRPC) Uplink(ctx context.Context, message *broker.Deduplica
 	if err := s.ValidateContext(ctx); err != nil {
 		return nil, errors.BuildGRPCError(err)
 	}
-	if !message.Validate() {
-		return nil, grpcErrf(codes.InvalidArgument, "Invalid Uplink")
+	if err := message.Validate(); err != nil {
+		return nil, errors.BuildGRPCError(errors.NewErrInvalidArgument("Uplink", err.Error()))
```

Comment:
Be consistent with `errors.Wrap` vs wrapping it in `errors.NewErrInvalidArgument`